meta/Decays.m

```diff
@@ -1104,18 +1129,28 @@ would violate Ward identity (as at the level of dim 4
            assignments
           ];
 
-(* replace 'physical_mass' with 'mass' to check Ward identity *)
+(* For tests: replace 'physical_mass' with 'mass' to check Ward identity *)
 FillSVVDecayAmplitudeMasses[decay_FSParticleDecay, modelName_, structName_, paramsStruct_] :=
     Module[{assignments = ""},
            assignments = assignments <> structName <> ".m_decay = " <> paramsStruct <> ".physical_mass<" <>
                          CXXDiagrams`CXXNameOfField[GetInitialState[decay]] <>
                          ">(idx_1);\n";
-           assignments = assignments <> structName <> ".m_vector_1 = " <> paramsStruct <> ".physical_mass<" <>
-                         CXXDiagrams`CXXNameOfField[First[GetFinalState[decay]]] <>
-                         ">(idx_2);\n";
-           assignments = assignments <> structName <> ".m_vector_2 = " <> paramsStruct <> ".physical_mass<" <>
-                         CXXDiagrams`CXXNameOfField[Last[GetFinalState[decay]]] <>
-                         ">(idx_3);\n";
+           assignments = assignments <> structName <> ".m_vector_1 = " <>
+                         If[IsMassless[First[GetFinalState[decay]]],
+                            "0.",
+                            If[IsZBoson[First[GetFinalState[decay]]],
+                               "qedqcd.displayPoleMZ()",
+                               paramsStruct <> ".physical_mass<" <> CXXDiagrams`CXXNameOfField[First[GetFinalState[decay]]] <> ">(idx_2)"
```

Comment:
I'm wondering if it is guaranteed that the variable `idx_2` is always defined. (Maybe it is.)

Reply:
Yes. There are always exactly 3 arguments called `idx_1`, `idx_2` and `idx_3`.

Reply:
Ah, ok!